src/dnx/dnx.cpp

```diff
@@ -2,106 +2,81 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 #include "stdafx.h"
+#include <vector>
 #include "pal.h"
 #include "utils.h"
 #include "app_main.h"
 
-bool LastIndexOfCharInPath(LPCTSTR pszStr, TCHAR c, size_t* pIndex)
+bool strings_equal_ignore_case(const dnx::char_t* s1, const dnx::char_t* s2)
 {
-    size_t nIndex = _tcsnlen(pszStr, MAX_PATH) - 1;
-    for (; nIndex != 0; nIndex--)
-    {
-        if (pszStr[nIndex] == c)
-        {
-            break;
-        }
-    }
-
-    *pIndex = nIndex;
-    return pszStr[nIndex] == c;
+#if defined(_WIN32)
+    return _wcsicmp(s1, s2) == 0;
+#else
+    return strcasecmp(s1, s2) == 0;
+#endif
 }
 
-bool PathEndsWith(LPCTSTR pszStr, LPCTSTR pszSuffix)
+bool string_ends_with_ignore_case(const dnx::char_t* s, const dnx::char_t* suffix)
 {
-    size_t nStrLen = _tcsnlen(pszStr, MAX_PATH);
-    size_t nSuffixLen = _tcsnlen(pszSuffix, MAX_PATH);
+    auto str_len = x_strlen(s);
+    auto suffix_len = x_strlen(suffix);
 
-    if (nSuffixLen > nStrLen)
+    if (suffix_len > str_len)
     {
         return false;
     }
 
-    size_t nOffset = nStrLen - nSuffixLen;
-
-    return ::_tcsnicmp(pszStr + nOffset, pszSuffix, MAX_PATH - nOffset) == 0;
+    return strings_equal_ignore_case(s + str_len - suffix_len, suffix);
 }
 
-bool LastPathSeparatorIndex(LPCTSTR pszPath, size_t* pIndex)
+int split_path(const dnx::char_t* path)
```

Comment:
Couldn't come up with a better name but if you have a proposal let me know... I am also still considering changing it to return two std::strings - directory and filename - these buffer operations are so fragile.